primitives/storage/storage.go

```diff
@@ -27,12 +29,29 @@ func Exists(key []byte) int32 {
 	return env.ExtStorageExistsVersion1(keyOffsetSize)
 }
 
-func Get(key []byte) []byte {
-	keyOffsetSize := utils.BytesToOffsetAndSize(key)
-	valueOffsetSize := env.ExtStorageGetVersion1(keyOffsetSize)
-	offset, size := utils.Int64ToOffsetAndSize(valueOffsetSize)
-	value := utils.ToWasmMemorySlice(offset, size)
-	return value
+func Get(key []byte) sc.Option[sc.Sequence[sc.U8]] {
```

Comment:
Wouldn't it be better if `Get,GetDecode,TakeBytes,TakeDecode` are part of the `system` module since the `storage` module is lower level and these functions are abstraction above it?

Reply:
I don't think they should be part of `system`. Maybe we can create an util package that wraps the low level calls.

Reply:
ok, not sure if the `system` module is an appropriate place, but I think it is worth it if we can keep the lower level functionality (residing in `/primitives/storage, hashing, ... etc`, interacting directly with the `wasm` `memory/storage` and things provided from the `/env`) separate from the things that work with higher level primitive types